okhttp/src/main/java/com/squareup/okhttp/CertificatePinner.java

```diff
@@ -123,17 +123,45 @@ private CertificatePinner(Builder builder) {
    * pinned for {@code hostname}. OkHttp calls this after a successful TLS
    * handshake, but before the connection is used.
    *
+   * @return {@code true} if hostname has pinned certificates and check passed,
+   * {@code false} if hostname doesn't have pinned certificates
    * @throws SSLPeerUnverifiedException if {@code peerCertificates} don't match
    *     the certificates pinned for {@code hostname}.
    */
-  public void check(String hostname, List<Certificate> peerCertificates)
+  public boolean check(String hostname, List<Certificate> peerCertificates)
       throws SSLPeerUnverifiedException {
-    List<ByteString> pins = hostnameToPins.get(hostname);
-    if (pins == null) return;
+
+    List<ByteString> directPins = hostnameToPins.get(hostname);
+    List<ByteString> wildcardPins = null;
+
+    int indexOfFirstDot = hostname.indexOf('.');
+
+    // Skip hostNames with one dot symbol for wildcard pattern search
+    //   example.com   will  be skipped
+    //   a.example.com won't be skipped
+    if (indexOfFirstDot != hostname.lastIndexOf('.')) {
```

Comment:
extract a field like `indexOfLastDot` for symmetry? Right now this is not as intuitive.